spec/index.bs

```diff
@@ -281,18 +281,18 @@ interface GPUShaderStageBit {
 
 enum GPUBindingType {
     "uniform-buffer",
-    "dynamic-uniform-buffer",
-    "sampler",
-    "sampled-texture",
     "storage-buffer",
-    "dynamic-storage-buffer"
+    "readonly-storage-buffer",
+    "sampler",
+    "sampled-texture"
     // TODO other binding types
 };
 
 dictionary GPUBindGroupLayoutBinding {
     required u32 binding;
     required GPUShaderStageFlags visibility;
     required GPUBindingType type;
+    bool dynamic = false;
```

Comment:
The title of the change does not mention anything about this flag.  What does 'dynamic' mean in this context?  Please add comments in the IDL or a section in the document that explains.

Reply:
Done. This was changed to address @kvark's concern that we have a combinatorial explosion of binding types.

Reply:
Thank you for the clarification, @Kangz .  Why does the implementation need to know whether a binding is dynamic or not?  Will it be a validation error if you set "dynamic = false" and use a dynamic offset anyways?

Reply:
Implementation needs to know if it expects that extra offset at the binding time. For example, in Vulkan it would use a different [VkDescriptorType](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkDescriptorType.html). The validation would need to make sure that exactly as many offsets are provided as there are dynamic buffers in the bind group.